### PR TITLE
2.1 Updates and Docker publishing

### PR DESCRIPTION
**Pull request by Drewpeifer**
### Contributions
* Many thanks to github users heehoo98 (contributed initial dockerization files), and user jzucker2 (more code and valuable advice on hosting a public docker image), latest should be available at `ghcr.io/drewpeifer/medialytics:latest`

### General
* Lots of cleanup, organization, and inline comments
* Renamed many variables for consistency, hopefully it's a little easier to follow now
* Converted chart colors to variables in `scripts.js` in case you want to change them or add additional colors

### Bugs Fixed
* Limits were returning 1 less than intended when rendering certain axes of charts

### New Features
* Some charts can now toggle between bar and pie by clicking the bar/pie icon in the top of their card
* Some charts now allow their limit (Top "X") value to be edited within the UI. If a limit longer than the list is entered, the total length of the list is used. Extremely large values may not display as intended (which is why there are limits by default, but they can be changed to any value, e.g. 999999, at  the top of `scripts.js`)
* Now tracking dateAdded (to library) and displaying "Earliest" and "Latest" addition
* Now tracking unmatched content and displaying it as a library statistic if any are found

### CODE REVIEW

**Comment by Drewpeifer** [comment]
@jzucker2whenever you have a few minutes to a look, this is very close to your PR I believe but I tried to set it up myself so that I understood what was going on a little better. Please feel free to let me know if you see any issues, it looks like the action was successful though and i see the container at ghcr.io/drewpeifer/medialytics:pr-7.  Thanks again!

**Comment by jzucker2** [comment]
Sounds good! I should be able to check it out today

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    directory: "/"
+    # Check for updates once a week
+    schedule:
+      interval: "daily"
+    reviewers:
+      - drewpeifer
+    assignees:
+      - drewpeifer
+    labels:
+      - github_actions
```
Comment (jzucker2) [inline .github/dependabot.yml:27]: So we should create these labels `github_actions`, it's on this page:
https://github.com/Drewpeifer/medialytics/labels

Or we should drop these lines. I usually create them, I find it helpful

**Review by jzucker2 — COMMENTED** [review]
Would it be ok if I made a commit or 2 directly to this branch? Would you like me to open a pr against this branch? I have a few quick comments below and I'm happy to jump in and do it or let you take you lead.

A few quick notes:

* I'd suggest updating the `docker-compose.yml` similar to how I did it in my PR. The main thing is it supports both local dev and install using the GitHub container registry
* Adding the `.dockerignore` as well (helps keep the docker container slim although the current `Dockerfile` shouldn't be an issue, just in case things change, it's nice to have)
* I also had some updates in the docs around this on your README
* We will also need a GitHub PR Label for the `github_actions` label I stuck in that dependabot file, happy to show you that, just wanted to communicate that. Or, we can drop those lines, I left a comment.

**Comment by jzucker2** [comment]
Also, last piece, once you start supporting Docker images in a container registry, you should probably try to be holistic about tagging/versioning because people tend to rely on that for Docker.

And we can try to add it to DockerHub too if you'd like, though I'd recommend doing that in a separate PR.

FYI the dependabot will open up PRs against your repo for updates to GitHub actions too. I believe my files are a version behind, we would see this on merge. I can also help you fix it too. We can adjust all of this.

**Comment by Drewpeifer** [comment]
@jzucker2 All of that makes sense to me, thanks for taking a look offering to edit! Please feel free to commit any changes directly against `develop` branch and they'll update on this PR. Whenever you're comfortable with the docker changes, you can comment or approve this PR, and I'll squash+merge after I review later (i might push a few small UI fixes this evening too).

For versioning, I'd like to be more consistent so any suggestions/edits there are welcome. There is a draft release for tag=v2.1 created.

**Comment by jzucker2** [comment]
> @jzucker2 All of that makes sense to me, thanks for taking a look offering to edit! Please feel free to commit any changes directly against `develop` branch and they'll update on this PR. Whenever you're comfortable with the docker changes, you can comment or approve this PR, and I'll squash+merge after I review later (i might push a few small UI fixes this evening too).
> 
> For versioning, I'd like to be more consistent so any suggestions/edits there are welcome. There is a draft release for tag=v2.1 created.

Pushed up a commit to follow up on my suggestions!

So usually semantic versioning is more like `v2.0.0` and this could be `v2.1.0` but that can also work. There's just no tag history prior to the first docker image, so going forward, we should make sure to create a tag for any big merges to master, basically, so that there can corresponding docker images.

Thanks so much for all the work on this project! I think it's really cool and I'm glad to have it running at my place.

**Pull request merged**